src/rust/en.omegascans/src/parser.rs

```diff
@@ -0,0 +1,406 @@
+
+	let url = format!("{}/series/{}", base_url, slug);
+
+	let status = data
+		.get("status")
+		.as_string()
+		.expect("Failed to get manga string as array")
+		.read();
+
+	let mut manga_status = MangaStatus::Unknown;
+
+	match status.as_str() {
+		"New" => {}
+		"Ongoing" => manga_status = MangaStatus::Ongoing,
+		"Completed" => manga_status = MangaStatus::Completed,
+		"Cancelled" => manga_status = MangaStatus::Cancelled,
+		"Dropped" => manga_status = MangaStatus::Cancelled,
+		"Hiatus" => manga_status = MangaStatus::Hiatus,
+		_ => manga_status = MangaStatus::Unknown,
+	};
+
+	Ok(Manga {
+		id: slug,
+		cover: cover,
+		title: title,
+		author: author,
+		artist: artist,
+		description: description,
+		url: url,
+		categories: Vec::new(),
+		status: manga_status,
+		nsfw: MangaContentRating::Nsfw,
+		viewer: MangaViewer::Scroll,
+	})
+}
+
+pub fn parse_chapter_list(base_url: String, manga_id: String) -> Result<Vec<Chapter>> {
+
+	let url = format!("https://api.omegascans.org/series/{}", manga_id);
+
+	let data = Request::new(url, HttpMethod::Get)
+		.json()
+		.expect("Failed to load JSON")
+		.as_object()
+		.expect("Failed to get JSON as object");
+
+	let mut all_chapters: Vec<Chapter> = Vec::new();
+
+	let seasons = data.get("seasons").as_array().expect("Failed to get seasons as array");
+
+	for season in seasons {
+		let season = season.as_object().expect("Failed to get season as object");
+		let volume = season.get("index").as_int().expect("Failed to get name as str");
+		
+		let chapters = season.get("chapters").as_array().expect("Failed to get chapters as array");
+
+		for chapter in chapters {
+			let chapter = chapter.as_object().expect("Failed to get chapter as object");
+
+			let price = chapter.get("price").as_int().expect("Failed to get chapter price");
+
+			if price != 0 {
+				continue;
+			}
+
+			let name = chapter.get("chapter_name").as_string().expect("Failed to get chapter name as str").read();
+			let index = chapter.get("index").as_string().expect("Failed to get chapter index as int").read();
+			let chapter_slug = chapter.get("chapter_slug").as_string().expect("Failed to get chapter slug as str").read();
+			let url = format!("{}/series/{}/{}", base_url, manga_id, chapter_slug);
+			let date_updated = chapter.get("created_at").as_string().expect("").read();
+			//aidoku::prelude::println!("{}", date_updated);
+			let date_updated = get_date(date_updated);
+
+			all_chapters.push(Chapter {
+				id: chapter_slug,
+				title: name,
+				volume: volume as f32,
+				chapter: index.parse::<f32>().unwrap_or_default(),
+				date_updated: date_updated,
+				scanlator: String::new(),
+				url: url,
+				lang: String::from("en"),
+			});
+			
+			
+
+		}
+	}
+		
+
+	Ok(all_chapters)
+}
+
+pub fn parse_page_list(base_url: String, manga_id: String, chapter_id: String) -> Result<Vec<Page>> {
+	let url = format!("{}/series/{}/{}", base_url, manga_id, chapter_id);
+
+	let obj = Request::new(url.as_str(), HttpMethod::Get).html()?;
```

Comment:
is `.as_str()` necessary? if so, you could probably just use `&` instead which is simpler